WorkbenchProject/src/main/java/edu/uwb/braingrid/workbench/provvisualizer/controller/ProVisCtrl.java

```diff
@@ -153,6 +155,19 @@ public void handle(long now) {
         };
 
         timer.start();
+
+        // Add an event handler to the primary stage to stop the timer when the application is minimized (iconified),
+        // and start it again when it is no longer minimized.
+        Stage primaryStage = WorkbenchDisplay.getPrimaryStage();
+        primaryStage.iconifiedProperty().addListener(
+                (observable, oldValue, newValue) -> {
+                    if (newValue) {
+                        timer.stop();
+                    } else {
+                        timer.start();
+                    }
+                }
+        );
```

Comment:
I'd prefer if this matched the more compact style of the other listeners in this class. Not a big deal, but I find the more compact version easier to read, especially when there are many of these statements in close proximity.
```suggestion
        primaryStage.iconifiedProperty().addListener((observable, oldValue, newValue) -> {
            if (newValue) {
                timer.stop();
            } else {
                timer.start();
            }
        });
```

Reply:
I see what you mean, I like consistency so that can be done!